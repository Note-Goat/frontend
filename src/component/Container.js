import Header from "./Header";
import {useContext} from "react";
import Context from "../Context";
import AuthedNav from "../app/nav/AuthedNav";
import UnAuthedNav from "../app/nav/UnAuthedNav";
import Footer from "./Footer";

export default function Container({ showNavigation, title, children }) {
  const { loggedIn } = useContext(Context);

  return (
    <div className="App">
      <Header title={title} />
      <div className="container">
        {(showNavigation || showNavigation === undefined) && (
          <div className="left-column">
            { loggedIn ? <AuthedNav /> : <UnAuthedNav /> }
          </div>
        )}
        <div className="right-column" style={{width: showNavigation ? "80%" : "100%"}}>
          {children}
        </div>
      </div>
      <Footer />
    </div>
  );
}
