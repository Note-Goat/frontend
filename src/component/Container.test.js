import renderer from 'react-test-renderer';
import Container from "./Container";
import {BrowserRouter} from "react-router-dom";

it('Container component snapshot', () => {
  const tree = renderer
    .create(<BrowserRouter><Container /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Container without navigation component snapshot', () => {
  const tree = renderer
    .create(<BrowserRouter><Container showNavigation={false} /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Container with title component snapshot', () => {
  const tree = renderer
    .create(<BrowserRouter><Container title="This is a title" /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
