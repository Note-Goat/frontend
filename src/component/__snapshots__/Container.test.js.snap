// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component snapshot 1`] = `
<div
  className="App"
>
  <div>
    <header
      className="App-header"
    >
      <img
        className="logo"
        src="/goat-original.png"
      />
      <h1
        className="header"
      />
    </header>
  </div>
  <div
    className="container"
  >
    <div
      className="left-column"
    >
      <nav>
        <ul
          className="nav"
        >
          <li>
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li>
            <a
              href="/signup"
              onClick={[Function]}
            >
              Signup
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="right-column"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="footer-container"
  >
    <div
      className="footer-controls"
    >
      <a
        href="/privacy"
        onClick={[Function]}
      >
        Privacy
      </a>
      <a
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
    </div>
  </div>
</div>
`;

exports[`Container with title component snapshot 1`] = `
<div
  className="App"
>
  <div>
    <header
      className="App-header"
    >
      <img
        className="logo"
        src="/goat-original.png"
      />
      <h1
        className="header"
      >
        This is a title
      </h1>
    </header>
  </div>
  <div
    className="container"
  >
    <div
      className="left-column"
    >
      <nav>
        <ul
          className="nav"
        >
          <li>
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li>
            <a
              href="/signup"
              onClick={[Function]}
            >
              Signup
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="right-column"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="footer-container"
  >
    <div
      className="footer-controls"
    >
      <a
        href="/privacy"
        onClick={[Function]}
      >
        Privacy
      </a>
      <a
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
    </div>
  </div>
</div>
`;

exports[`Container without navigation component snapshot 1`] = `
<div
  className="App"
>
  <div>
    <header
      className="App-header"
    >
      <img
        className="logo"
        src="/goat-original.png"
      />
      <h1
        className="header"
      />
    </header>
  </div>
  <div
    className="container"
  >
    <div
      className="right-column"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="footer-container"
  >
    <div
      className="footer-controls"
    >
      <a
        href="/privacy"
        onClick={[Function]}
      >
        Privacy
      </a>
      <a
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
    </div>
  </div>
</div>
`;
